presto-hive/src/main/java/com/facebook/presto/hive/util/AsyncQueue.java

```diff
@@ -208,6 +208,19 @@ else if (finishing && borrowerCount == 0) {
                 }, directExecutor());
     }
 
+    public int size()
+    {
+        return elements.size();
+    }
+
+    public void clear()
+    {
+        elements.clear();
+        notFullSignal = SettableFuture.create();
```

Comment:
Do we want to cancel the current `notFullSignal` and `notEmptySignal` signal? :)

Reply:
No longer relevant given we decided to create new async queue instead of clear the queue :D